docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+3. ```ParsePair``` also calls helper methods in ```PairingList``` to check that the pairing client and property indexes
+    exists. Also, the client and property must not be already paired. The client must not be renting any property
+    presently as well.
+4. After passing all these checks, the program fetches the desired```Property``` and ```Client``` from
+   ```PropertyList``` and ```ClientList```.
+5. The ```Property``` and ```Client``` objects are inserted as a pair into the hashmap of ```PairingList```.
+ 
+#### Unpair
+
+The unpair command takes in user input of the format:
+```
+unpair ip/PROPERTY_INDEX ic/CLIENT_INDEX
+```
+where ```PROERTY_INDEX``` and ```CLIENT_INDEX``` must be positive integers which are indexes present in ```ClientList```
+and ```PropertyList``` if their private arrays were 1-indexed.
+
+
+How the unpair command works:
+1. The user input for a pair command is first parsed by ```Parser``` (specifically, ```ParseUnpair```).
+2. ```ParseUnpair``` checks the user input for formatting mistakes such as missing flags and wrong flag orders.
+3. ```ParseUnpair``` also calls helper methods in ```PairingList``` to check that the pairing client and property
+   indexes exist, and that the client-property pair exist in ```PairingList```.
+4. After passing all these checks, the ```PairingList``` deletes the hashmap entry in ```clientPropertyPairings```
+   which contains the client-property pair.
+
+---
+
+### Storage
+The implementation of Storage class requires consists of different level of operations:
+
+- Load Files
+- Append to File
+- Update to File
+
+#### Load Files:
+![Load File Flowchart](diagrams/LoadFileFlowChart.jpg)
+At the file loading level, it comprises checks to verify the directory is created. This is done by invoking a method: 
+`loadFiles(hasDirectory, hasPropertyFile, hasClientFile, hasPairingFile, clientList, propertyList, pairingList)`.
+This method would conduct the following operations:
+- Create a `data` directory if not already exist. (`hasDirectory` is `false`)
+- Load Clients into Client ArrayList if `hasClientFile` is `true`.
+- Load Properties into Property ArrayList if `hasPropertyFile` is `true`.
+- Load Pairings into Pairing HashMap if `hasPairingFile` is `true`.
+
+An empty file would not be loaded into the ArrayList and PropertyList as the code is designed to read for `next()`.
+An empty file would invoke a `false` in `hasNext()`, thus adding operation would not continue. The overall operation can
+be visualised in the flowchart above.
+
+#### Append To File
+When file is appended into the text file, it's being stored in different formats as shown below:
+
+- Client:  `NAME | CONTACT_NUMBER | EMAIL <optional> | BUDGET` 
+- Property: `NAME | ADDRESS | RENTAL_PRICE | UNIT_TYPE` 
+- Pairing: `[CLIENT_FORMAT] : [PROPERTY_FORMAT]` 
+
+The text file of which Client, Property and Pairing is being stored is `client.txt`, `property.txt` and `pairing.txt` 
+respectively.
+
```

Comment:
I think all the method calls in the diagram can be more readable by dropping the "this."